common/djangoapps/contentserver/middleware.py

```diff
@@ -159,18 +185,37 @@ def set_caching_headers(self, content, response):
         force_header_for_response(response, 'Vary', 'Origin')
 
     @staticmethod
+    def is_cdn_request(request):
+        """
+        Attempts to determine whether or not the given request is coming from a CDN.
+
+        Currently, this is a static check because edx.org only uses CloudFront, but may
+        be expanded in the future.
+        """
+        user_agent = request.META.get('HTTP_USER_AGENT', '')
+        if user_agent == 'Amazon CloudFront':
+            # This is a CloudFront request.
+            return True
+
+        return False
```

Comment:
Can we make this configurable? I think this would be the first thing we'd add that was explicitly hard-coded to CloudFront.

Nitpick: the last part is equivalent to just `return (user_agent == 'Amazon CloudFront')`